cheri_compressed_cap_common.h

```diff
@@ -67,22 +67,32 @@ _CC_STATIC_ASSERT_SAME(_CC_N(FIELD_INTERNAL_EXPONENT_SIZE) + _CC_N(FIELD_EXP_NON
                            _CC_N(FIELD_EXPONENT_LOW_PART_SIZE),
                        _CC_N(FIELD_EBT_SIZE));
 
+// In order to allow vector loads and store from memory we can optionally reverse the first two fields.
 struct _cc_N(cap) {
     /* offset = cursor - base */
+#ifdef _CC_REVERSE_PESBT_CURSOR_ORDER
+    /* Currently only updated lazily when compress is called.
+     * We might want to make this the canonical field and slice out other fields on demand.*/
+    uint64_t cached_pesbt;
     _cc_addr_t _cr_cursor; /* Capability cursor */
-    _cc_addr_t cr_base;    /* Capability base addr */
+#else
+    _cc_addr_t _cr_cursor;
+    _cc_addr_t cached_pesbt;
+#endif
     _cc_length_t _cr_top;  /* Capability top */
+    _cc_addr_t cr_base;    /* Capability base addr */
     uint32_t cr_perms;     /* Permissions */
     uint32_t cr_uperms;    /* User Permissions */
```

Comment:
Considering the awkward differences between morello+non-Morello, I wonder if we should just merge these fields or drop them entirely and always extract from pesbt.

Reply:
I'm fine with merging, the distinction doesn't really make much sense here IMO and the current representation is pretty wasteful of space. Probably nicer to keep it as a separate field though to match everything else.

Reply:
+1 Merge. Unsure about removing entirely. Would have to profile to be sure, but my guess is that the field is read a lot more than written. Should give fewer TCG ops overall.

Reply:
Although I think I will leave merging the fields for a later refactor.